### PR TITLE
Implement loader for DEX parsing

### PR DESCRIPTION
**Pull request by CharlesJ10**

### CODE REVIEW

**Review by Chronos34 — APPROVED** [review]

**Review by ziniblesssed — APPROVED** [review]
LGTM

File: capstone/src/main/java/com/google/sps/data/dexparser/DexLoader.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    try {
+
+      //Declare unzip elements
+      FileInputStream is = new FileInputStream(filePath);
+      ZipInputStream zis = new ZipInputStream(is);
+      ZipEntry ze = zis.getNextEntry();
+
+      while(ze != null) {
+        String fileName = ze.getName();
+        LOGGER.info("File Name: " + fileName);
```
Comment (viksabnis) [inline capstone/src/main/java/com/google/sps/data/dexparser/DexLoader.java:53]: string formatting can be used

Comment (CharlesJ10) [reply capstone/src/main/java/com/google/sps/data/dexparser/DexLoader.java:53]: Could you elaborate on it?

**Comment by CharlesJ10** [comment]
@googlebot I fixed it.

File: capstone/src/main/java/com/google/sps/data/dexparser/DexLoader.java (hunk 1 above)
Comment (viksabnis) [reply capstone/src/main/java/com/google/sps/data/dexparser/DexLoader.java:53]: LOGGER.log(Level.INFO,"Hello World {0}", "Param");
But that might be too much

File: capstone/src/test/java/com/google/sps/DexLoaderTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  /*
+   * Tests that loading of DEX parser is unsuccessful with incorrect file path
+   */
+  @Test
+  public void analyzeApkFeaturesLocallyTestWithIncorrectFilePath() {
+
+    DexLoader dexLoader = new DexLoader();
+
+    // To run this locally, change the file path to your current one
+    String filePath = "/home/charlesogbogu/step2-2020/capstone/src/main/resources/IncorrectHelloActivity.apk";
```
Comment (HallM) [inline capstone/src/test/java/com/google/sps/DexLoaderTest.java]: If this test is supposed to be a file that doesn't exist, then the filePath could be something more placeholder looking like "/path/to/FileThatDoesNotExist.apk".

File: capstone/src/main/java/com/google/sps/data/dexparser/DexLoader.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        }
+        //close this ZipEntry
+        zis.closeEntry();
+        ze = zis.getNextEntry();
+      }
+
+      System.out.println();
+
+      //Print the feature to the console
+      LOGGER.info("" + filesCount);
```
Comment (HallM) [inline capstone/src/main/java/com/google/sps/data/dexparser/DexLoader.java]: Outputting what the number means is helpful when looking at logs. Either could be
LOGGER.info("File Count: " + filesCount);
or with the formatter style
LOGGER.info("File Count: {0}", filesCount);

Same with below.

**Review by HallM — CHANGES_REQUESTED** [review]

File: capstone/src/test/java/com/google/sps/DexLoaderTest.java (hunk 2 above)
Comment (viksabnis) [reply capstone/src/test/java/com/google/sps/DexLoaderTest.java]: +1

File: capstone/src/main/java/com/google/sps/data/dexparser/DexLoader.java (hunk 3 above)
Comment (viksabnis) [reply capstone/src/main/java/com/google/sps/data/dexparser/DexLoader.java]: same comment as earlier - though 
LOGGER.log(Level.INFO, "File Count: {0}", filesCount);

**Review by HallM — APPROVED** [review]

**Review by viksabnis — APPROVED** [review]

**Pull request merged**